### PR TITLE
Create HAT Diamond [keep this for reference]

### PR DESCRIPTION
**Pull request by ben-kaufman**
Use the Diamond pattern for the HATVaults contract

### CODE REVIEW

File: scripts/hatvaultsdeploy.js [hunk 1]
```diff
@@ -1,80 +1,169 @@
...
     console.warn(
       "You are trying to deploy a contract to the Hardhat Network, which" +
         "gets automatically created and destroyed every time. Use the Hardhat" +
         " option '--network localhost'"
     );
   }
 
-  const addresses = ADDRESSES[network.name];
+  const diamond = require("./diamond-util.js");
+  let totalGasUsed = ethers.BigNumber.from("0");
```
Comment (jellegerbrandy) [inline scripts/hatvaultsdeploy.js:22]: ?

File: contracts/facets/DepositFacet.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
```
Comment (jellegerbrandy) [inline contracts/facets/DepositFacet.sol:1]: seems a bit weird to me to have a deposit and a separate reward facet, let's merge thos

File: contracts/facets/CommitteeFacet.sol [hunk 3]
```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
```
Comment (jellegerbrandy) [inline contracts/facets/CommitteeFacet.sol:2]: Seems to me the "setFoo" functions should in the paramsfacet, right?

**Review by jellegerbrandy — CHANGES_REQUESTED** [review]
I think this is very good and clear, I left some small suggestions

Heads up that will ahve a tough time fixing merge conflicts later, so I did not check if all the code was propertly transfered yet, I'll do another review then

File: contracts/facets/CommitteeFacet.sol (hunk 3 above)
Comment (ben-kaufman) [reply contracts/facets/CommitteeFacet.sol:2]: Well these which are set by the committee I put here, and the general params set by hats governance I put in the params. Makes sense or what do you think?

File: contracts/facets/DepositFacet.sol (hunk 2 above)
Comment (ben-kaufman) [reply contracts/facets/DepositFacet.sol:1]: Ok merged them

File: scripts/hatvaultsdeploy.js (hunk 1 above)
Comment (ben-kaufman) [reply scripts/hatvaultsdeploy.js]: Removed

**Pull request closed without merging**